extension/compile.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+cat "../src/SCHEMA.sql" | grep -v 'DROP SCHEMA' | grep -v 'CREATE SCHEMA'
+
+scripts=("SETUP.sql" "LOG_UTIL.sql" "DDL_LOG.sql" "RESTORE.sql" "REVERT.sql" "SCHEMA_MANAGEMENT.sql");
+
+for script in ${scripts[@]}; do
+    cat "../src/$script";
+    echo -e "\n\n\n";
+done
+
+
+echo "-- make all the data available for pg_dump";
+read -r -d '' CONFIG_DUMP <<'EOF'
+do language plpgsql
+$$
+declare
+  name_ varchar;
+begin
+  for name_ in select sequence_schema || '.' || sequence_name from information_schema.sequences where sequence_schema = 'pgmemento' loop
+    perform pg_catalog.pg_extension_config_dump(name_, '');
+  end loop;
+
+  for name_ in select table_schema || '.' || table_name from information_schema.tables where table_schema = 'pgmemento' loop
+    perform pg_catalog.pg_extension_config_dump(name_, '');
```

Comment:
Yeah, I've been thinking about 3 possible options
 1. create the data tables when user activates the extension (e.g. a new procedure for INIT.sql)
 2. perform `alter extension drop ...` for the data objects in the extension install script
 3. mark data tables as config so it's controlled by the extension but makes it to the pg_dump

I'm not sure at this point which one is the best. Each one has its own pros and cons, but 3rd seems to be the most seamless to me. If we keep the data completely detached from the extension, it might be counterintuitive for users that we could update their structure or delete them on the extension update (if we want to issue an extension update). If we drop them from the extension, we won't be able to `drop extension` without asking users to manually drop the tables first.
I feel like the `option 2` is clearly better than the `option 1`, but still not sure about `option 2` vs `option 3`.

Reply:
Well, don't think we should be afraid of a `drop extension` by an accident as long as the data is preserved in database dumps.
I would guess `drop database` by accident is another possible option, but we can't always protect users from themselves :)

Reply:
Not 100% sure what you mean with "data tables". The log tables inside the pgMemento schema? Or normal tables with the additional `audit_id` column?

If I would run `CREATE EXTENSION pgmemento;` I would expect the same behaviour like running the `INSTALL_PGMEMENTO`script. Then there could be an extra `init` function next to the current interactive version.

When calling `DROP EXTENSION pgmemento;` the log tables in the `pgmemento` schema could be saved in a dump, but the normal tables should stay (except for dropping triggers and audit_id column). Maybe with `CASCADE` there should be no dumps then.

Reply:
> what you mean with "data tables"

By `data tables` I'm talking about `pgmemento.*` tables keeping history (the log tables). Those are the only ones generated and controlled by the extension.

> When calling `DROP EXTENSION pgmemento;` the log tables in the `pgmemento` schema could be saved in a dump

Yeah, that's what troubles me. If the user runs `DROP EXTENSION` without taking the database dump first, all the `pgmemento.*` data (log) tables could be lost as the extension drops them all.

The user tables won't be touched of course, only the triggers could be deleted in case of cascade, but that's quite normal I believe.

Reply:
Ah, now I got it. Should have read the Postgres docs more carefully. First, I thought that `pg_catalog.pg_extension_config_dump` will automatically create a dump as soon as I user deletes the extension ^^

Then I would say option 3 is the most intuitive. I'm working a lot with PostGIS and whenever I drop it, I expect every PostGIS element to be gone afterwards. This is also the current behaviour of `UNINSTALL_PGMEMENTO.sql`.